features/include/pcl/features/flare.h

```diff
@@ -0,0 +1,300 @@
+          min_neighbors_for_tangent_axis_ = min_neighbors_for_tangent_axis;
+        }
+
+        /** \brief Get min number of neighbours required for the computation of X axis.
+        *
+        * \return min number of neighbours required for the computation of X axis.
+        */
+        inline int
+          getMinNeighboursForTangentAxis () const
+        {
+          return (min_neighbors_for_tangent_axis_);
+        }
+
+
+        /** \brief Provide a pointer to the dataset used for the estimation of X axis.
+        * As the estimation of x axis is negligibly affected by surface downsampling,
+        * this method lets to consider a downsampled version of surface_ in the estimation of x axis.  
+        * This is optional, if this is not set, it will only use the data in the
+        * surface_ cloud to estimate the x axis. 
+        * \param[in] cloud a pointer to a PointCloud 
+        */
+        inline void 
+          setSearchSampledSurface(const PointCloudInConstPtr &cloud)
+        {
+          sampled_surface_ = cloud;
+          fake_sampled_surface_ = false;
+        }
+
+        /** \brief Get a pointer to the sampled_surface_ cloud dataset. */
+        inline PointCloudInConstPtr 
+          getSearchSampledSurface() const
+        {
+          return (sampled_surface_);
+        }
+
+        /** \brief Provide a pointer to the search object linked to sampled_surface.
+        * \param[in] tree a pointer to the spatial search object linked to sampled_surface.
+        */
+        inline void 
+          setSearchMethodForSampledSurface (const KdTreePtr &tree) { sampled_tree_ = tree; }
+
+        /** \brief Get a pointer to the search method used for the extimation of x axis. */
+        inline KdTreePtr 
+          getSearchMethodForSampledSurface () const
+        {
+          return (sampled_tree_);
+        }
+
+        /** \brief Get the signed distances of the highest points from the fitted planes. */
+        inline std::vector<SignedDistanceT> & 
+          getSignedDistancesFromHighestPoints () 
+        {
+          return (signed_distances_from_highest_points_);
+        }
+
+  protected:
+    /** \brief This method should get called before starting the actual computation. */
+    virtual bool
+      initCompute ();
+
+    /** \brief This method should get called after the actual computation is ended. */
+    virtual bool
+      deinitCompute ();
+
+    /** \brief Estimate the LRF descriptor for a given point based on its spatial neighborhood of 3D points with normals
+    * \param[in] index the index of the point in input_
+    * \param[out] lrf the resultant local reference frame
+    * \return signed distance of the highest point from the fitted plane. Max if the lrf is not computable.
+    */
+    SignedDistanceT
+      computePointLRF (const int &index, Eigen::Matrix3f &lrf);
+
+    /** \brief Abstract feature estimation method.
+    * \param[out] output the resultant features
+    */
+    virtual void
```

Comment:
And that's why `override` appeared later on and we should try to adopt it once we migrate. But you're twisting the original purpose of the `virtual`keyword. I really think it should only be declared if one expects the classe to be derived later on, and that somehow one will be only working with methods which work with pointers to the base class, and do not care about the derived class.